modules/koblerBidAdapter.js

```diff
@@ -0,0 +1,226 @@
+import * as utils from '../src/utils.js';
+import {config} from '../src/config.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'kobler';
+const BIDDER_ENDPOINT = 'https://bid.essrtb.com/bid/prebid_rtb_call';
+const TIMEOUT_NOTIFICATION_ENDPOINT = 'https://bid.essrtb.com/notify/prebid_timeout';
+const DEFAULT_CURRENCY = 'USD';
+const DEFAULT_TIMEOUT = 1000;
+const TIME_TO_LIVE_IN_SECONDS = 10 * 60;
+
+export const isBidRequestValid = function (bid) {
+  return !(!bid || !bid.bidId || !bid.params || !bid.params.placementId);
+};
+
+export const buildRequests = function (validBidRequests, bidderRequest) {
+  return {
+    method: 'POST',
+    url: BIDDER_ENDPOINT,
+    data: buildOpenRtbBidRequestPayload(validBidRequests, bidderRequest),
+    options: {
+      contentType: 'application/json'
+    }
+  };
+};
+
+export const interpretResponse = function (serverResponse) {
+  const res = serverResponse.body;
+  const bids = []
+  if (res) {
+    res.seatbid.forEach(sb => {
+      sb.bid.forEach(b => {
+        bids.push({
+          requestId: b.impid,
+          cpm: b.price,
+          currency: res.cur,
+          width: b.w,
+          height: b.h,
+          creativeId: b.crid,
+          dealId: b.dealid,
+          netRevenue: true,
+          ttl: TIME_TO_LIVE_IN_SECONDS,
```

Comment:
Do we really need to set TTL ?

Reply:
I think so, but it's good because the token will expire in 10 minutes anyway and after that, we won't serve the ad.

Reply:
We can retain. One corner case is prefetching the ad. But 10 min should be fin.